flink-cdc-common/src/main/java/com/ververica/cdc/common/event/AddColumnEvent.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.event;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import com.ververica.cdc.common.schema.Column;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A {@link SchemaChangeEvent} that represents an {@code ADD COLUMN} DDL, which may contain the
+ * lenient column type changes.
+ */
+@PublicEvolving
+public final class AddColumnEvent implements SchemaChangeEvent {
+
+    private static final long serialVersionUID = 1L;
+
+    private final TableId tableId;
+
+    private final List<Column> addedColumns;
```

Comment:
When adding column, the column order may be changed. I don't see this information in this class. 
For example, alter column in Mysql.
```sql
ALTER TABLE table_name
ADD [COLUMN] column_definition [FIRST|AFTER existing_column]
```

Reply:
Add `ColumnWithPosition` to address it.

Reply:
Hi @ruanhang1993, I found that [columns in Schema](https://github.com/ververica/flink-cdc-connectors/blob/99d1ad5e528e9531ed1eb703479e7f064f910930/flink-cdc-common/src/main/java/com/ververica/cdc/common/schema/Schema.java#L48) is unordered. So it would be invalid for CreateTableEvent to set position.

Reply:
There will be another pr to solve the right order in Schema. So we can ignore it and continue to complete this pr.